app/src/main/java/com/example/vanna/attendance2/history/AttenddanceHistoryActivity.java

```diff
@@ -14,18 +18,51 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_attenddance_history);
 
+        initLayout();
+
         RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
 
         // setup adapter
         Attendance[] data = getAttendanceData();
-        AttendanceArrayAdapter adapter = new AttendanceArrayAdapter(data);
+        AttendanceArrayAdapter adapter = new AttendanceArrayAdapter(this, data);
```

Comment:
should use getApplicationContext() instead of this. "this" here refer to activity and you could leak memory using this